### DISCUSSION TITLE
Can't get WebRTC and HLS to work at the same time with a ffmpeg generated rtsp delivered stream

### DISCUSSION THREAD

**Discussion by thuusom** [bluenviron/mediamtx#3887] [Questions]
### Question

I have enabled hls and webrtc as well as rtsp (tcp/udp) on a local mediamtx 1.9.2. All is standard configuration.
When I run an ffmpeg generated stream that is served to the mediamtx on rtsp, depending on the options passed on the the commandline it either works well with webrtc or hls, but not well for both at the same time. Is that how its supposed to work?

To reproduce enable the three services and start mediamtx locally. Then try the two different ffmpeg cmd lines below with two browser windows opens (chrome in this case) on http://localhost:8888/cam1 and http://localhost:8889/cam1. Notice the difference. 

webrtc fails and hls works:
ffmpeg -f lavfi -i testsrc=size=640x480:rate=10 \
-vf "drawtext=fontfile=/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf:text='%{localtime}: %{fps} FPS':fontcolor=white:fontsize=24:x=10:y=10" \
-vcodec libx264 -pix_fmt yuv420p -r 30 -b:v 300k -f rtsp rtsp://localhost:8554/cam1

webrtc works and hls works but flacky:
ffmpeg -f lavfi -i testsrc=size=640x480:rate=10 \
-vf "drawtext=fontfile=/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf:text='%{localtime}: %{fps} FPS':fontcolor=white:fontsize=24:x=10:y=10" \
-vcodec libx264 -pix_fmt yuv420p -r 30 -tune zerolatency -preset ultrafast -b:v 300k -fflags nobuffer \
-f rtsp rtsp://localhost:8554/cam1

**Comment by aler9** [comment]
Both commands are wrong, for several reasons (B-frames, no -re mode and other things).

Suggested commands are in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#remuxing-re-encoding-compression
